adapters/akka-http/src/main/scala/caliban/AkkaHttpAdapter.scala

```diff
@@ -64,6 +52,33 @@ object AkkaHttpAdapter {
     )
   }
 
+  def makeHttpUploadService[R, E](
+    interpreter: GraphQLInterpreter[R, E],
+    skipValidation: Boolean = false,
+    enableIntrospection: Boolean = true,
+    queryExecution: QueryExecution = QueryExecution.Parallel,
+    requestInterceptor: RequestInterceptor[R] = RequestInterceptor.empty
+  )(implicit
+    runtime: Runtime[R with Random],
+    requestCodec: JsonCodec[GraphQLRequest],
+    mapCodec: JsonCodec[Map[String, Seq[String]]],
+    responseCodec: JsonCodec[GraphQLResponse[E]]
+  ): Route = {
+    val endpoint = TapirAdapter.makeHttpUploadService[R, E](
+      interpreter,
+      skipValidation,
+      enableIntrospection,
+      queryExecution,
+      requestInterceptor
+    )
+    AkkaHttpServerInterpreter().toRoute(
+      ServerEndpoint[(Seq[Part[Array[Byte]]], ServerRequest), StatusCode, GraphQLResponse[E], Any, Future](
+        endpoint.endpoint,
+        _ => req => runtime.unsafeRunToFuture(endpoint.logic(zioMonadError)(req)).future
+      )
+    )
+  }
```

Comment:
So we'll still need these sort of capabilities added for each of the adapters right?

Reply:
What do you mean?

Reply:
Like the `makeUploadService` we'll need to define that per adapter still?

Reply:
Ah yeah, but it's just a convenience alias. It just calls `makeUploadService` from `TapirAdapter` (for Akka/Play there is some extra code to convert between ZIO types and Future/Flow, which I should move to a public function). Users can still use `TapirAdapter` instead of that.